src/strat_engine/blockdev.rs

```diff
@@ -55,87 +55,109 @@ impl BlockDev {
-        }
+    /// Initialize multiple blockdevs at once. This allows all of them
+    /// to be checked for usability before writing to any of them.
+    pub fn batch_new(stratis_id: &str,
+                     paths: &[&Path],
+                     mda_size: Sectors,
+                     force: bool)
+                     -> EngineResult<Vec<BlockDev>> {
 
         if *mda_size % 2 != 0 {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is not an even \
+                                                      format!("mda size {} is not an even \
                                                                number",
-                                                              path.display()))));
+                                                              *mda_size))));
         }
 
         if mda_size < MIN_MDA_SIZE {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is less than \
+                                                      format!("mda size {} is less than \
                                                                minimum ({})",
-                                                              path.display(),
+                                                              *mda_size,
                                                               *MIN_MDA_SIZE))));
         }
 
-        let dev = try!(Device::from_str(&path.to_string_lossy()));
+        // Check stuff for all candidates before writing anything
+        let mut good_paths = Vec::new();
+        for path in paths {
+            let metadata = try!(fs::metadata(path));
 
-        let mut f = try!(OpenOptions::new()
-            .read(true)
-            .open(path)
-            .map_err(|_| {
-                io::Error::new(ErrorKind::PermissionDenied,
-                               format!("Could not open {}", path.display()))
-            }));
+            if !metadata.file_type().is_block_device() {
+                return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
+                                                          format!("{} is not a block device",
+                                                                  path.display()))));
+            }
 
-        if !force {
-            let mut buf = [0u8; 4096];
-            try!(f.read(&mut buf));
+            let dev = try!(Device::from_str(&path.to_string_lossy()));
+
+            let mut f = try!(OpenOptions::new()
+                .read(true)
+                .write(true)
+                .open(path)
+                .map_err(|_| {
+                    io::Error::new(ErrorKind::PermissionDenied,
+                                   format!("Could not open {}", path.display()))
+                }));
+
+            if !force {
```

Comment:
I reopened the issue about this. I'm pretty confident that we agreed not to force.